From: Matthew Garrett <matthew.garrett@nebula.com>
Subject: [PATCH V3 09/11] uswsusp: Disable when module loading is restricted
Date: Tue,  3 Sep 2013 19:50:16 -0400
Target: opensuse 13.1
Patch-mainline: fedora 12

uswsusp allows a user process to dump and then restore kernel state, which
makes it possible to avoid module loading restrictions. Prevent this when
any restrictions have been imposed on loading modules.

Signed-off-by: Matthew Garrett <matthew.garrett@nebula.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com>

---
 kernel/power/user.c | 4 ++++
 1 file changed, 4 insertions(+)

Index: linux-3.11-openSUSE-13.1/kernel/power/user.c
===================================================================
--- linux-3.11-openSUSE-13.1.orig/kernel/power/user.c
+++ linux-3.11-openSUSE-13.1/kernel/power/user.c
@@ -24,6 +24,7 @@
 #include <linux/console.h>
 #include <linux/cpu.h>
 #include <linux/freezer.h>
+#include <linux/module.h>
 
 #include <asm/uaccess.h>
 
@@ -48,6 +49,9 @@ static int snapshot_open(struct inode *i
 	struct snapshot_data *data;
 	int error;
 
+	if (secure_modules())
+		return -EPERM;
+
 	lock_system_sleep();
 
 	if (!atomic_add_unless(&snapshot_device_available, -1, 0)) {
